### PR TITLE
Link API docs and assorted fixes

### PR DESCRIPTION
**Pull request by elenadimitrova**
**Description**
A handful of fixes and linking in content from monorepo and website.

- Restore in navigation the missing "Bridge basics" page (probably gone when rebasing Kelvin's dev docs PR)
- Link the new API docs [INT-456](https://linear.app/optimism/issue/INT-456/link-api-docs)
- Remove RPGF content as this is added to the website
- Remove dedicated contact page and add content to homepage instead
- Add new root section for Guides
- ~Remove protocol information which is out of date. I've just kept the "sequencing" page to light-paper section where it might be useful for this new future section.~
- Remove obsoleted assets for the user docs content

**Metadata**
- Fixes INT-456

### CODE REVIEW

File: src/.vuepress/config.js [hunk 1]
```diff
@@ -89,22 +98,10 @@ module.exports = {
         ['https://www.optimism.io/apps/tools', 'Third Party Tools'],
       ],
       '/docs/protocol/': [
-        '/docs/protocol/protocol-2.0.md',
```
Comment (smartcontracts) [inline src/.vuepress/config.js:92]: Is the plan to add new content to this section in the future?

Comment (smartcontracts) [reply src/.vuepress/config.js:92]: Hmm that could make sense. I was hoping to still have a section of the docs that could be an approachable representation of the protocol vs. low-level specs. Are we still thinking about having a section like that?

Comment (elenadimitrova) [reply src/.vuepress/config.js:92]: Yes there is the light paper section called "How it works" which is the first section in the new designs.

Comment (elenadimitrova) [reply src/.vuepress/config.js]: I've dropped that commit and reverted to the original protocol content. I'll let John do that in a separate PR

**Pull request merged**